main.go

```diff
@@ -18,5 +18,223 @@
+								break
+							}
+							updateConnections(logger, nsmgrAddr, event)
+						}
+					}()
+				}
+			}
+		}
+	}
+}
+
+func getNseChannel(ctx context.Context, logger log.Logger, dialOptions []grpc.DialOption) <-chan *registry.NetworkServiceEndpointResponse {
+	conn, err4 := grpc.Dial(registryAddr, dialOptions...)
+	if err4 != nil {
+		logger.Errorf("Failed dial to Registry: %v", err4.Error())
+	}
+
+	clientNse := registry.NewNetworkServiceEndpointRegistryClient(conn)
+
+	streamNse, err5 := clientNse.Find(ctx, &registry.NetworkServiceEndpointQuery{
+		Watch: true,
+		NetworkServiceEndpoint: &registry.NetworkServiceEndpoint{
+			NetworkServiceNames: []string{"forwarder"},
+		},
+	})
+	if err5 != nil {
+		logger.Errorf("Failed to perform Find NSE request: %v", err5)
+	}
+
+	return registry.ReadNetworkServiceEndpointChannel(streamNse)
+}
+
+func updateConnections(logger log.Logger, nsmgrAddr string, event *networkservice.ConnectionEvent) {
+	for _, connection := range event.Connections {
+		connectionID := generateConnectionID(connection)
+		logger.Infof("Handling %q event on %q nsmgr connection: %q", event.Type.String(), nsmgrAddr, connectionID)
+		switch {
+		case event.Type == networkservice.ConnectionEventType_DELETE:
+			delete(connections, connectionID)
+			removeManagerConnection(nsmgrAddr, connectionID)
+		default: // INITIAL_STATE_TRANSFER and UPDATE event types
+			connections[connectionID] = connection
+			addManagerConnection(nsmgrAddr, connectionID)
+		}
+	}
+	parceConnectionsToGraphicalModel()
+}
+
+func cleanupManager(logger log.Logger, nsmgrAddr string) {
+	for connectionID := range managerConnections[nsmgrAddr] {
+		logger.Infof("Removing %q nsmgr connection: %q", nsmgrAddr, connectionID)
+		delete(connections, connectionID)
+	}
+	delete(managerConnections, nsmgrAddr)
+	delete(nsmgrs, nsmgrAddr)
+	parceConnectionsToGraphicalModel()
+}
+
+func configureAndRunRESTServer(logger log.Logger) {
+	r := mux.NewRouter()
+
+	r.HandleFunc("/nodes", getNodesHandler).Methods("GET")
+	r.HandleFunc("/edges", getEdgesHandler).Methods("GET")
+
+	http.Handle("/", r)
+	go func() {
+		server := &http.Server{
+			Addr:         ":3001",
+			Handler:      nil,
+			ReadTimeout:  10 * time.Second,
+			WriteTimeout: 10 * time.Second,
+		}
+
+		if err3 := server.ListenAndServe(); err3 != nil {
+			logger.Fatalf("Error starting server: %v", err3)
+		}
+	}()
+}
+
+func getNodesHandler(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("Access-Control-Allow-Origin", "http://localhost:3000")
+	err := json.NewEncoder(w).Encode(storageData.Nodes)
+	if err != nil {
+		handleError(w, err)
+		return
+	}
+}
+
+func getEdgesHandler(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("Access-Control-Allow-Origin", "http://localhost:3000")
+	err := json.NewEncoder(w).Encode(storageData.Edges)
+	if err != nil {
+		handleError(w, err)
+		return
+	}
 }
```

Comment:
Why we are not using gin framework as was planned?

Reply:
implemented via gin framework